### DISCUSSION TITLE
Custom /api path

### DISCUSSION THREAD

**Discussion by alexander-dmitriev** [vercel/next.js#38669] [Ideas]
### Describe the feature you'd like to request

I make localized documentation site with dynamic urls like `https://site/locale/theme/article`.
But when I open url `https://site/en/api/methods` I get a 404 error, because /api path reserved and cannot be changed. There is no option to rename the theme. However I want to use the API Routes in this project.

How can I get the urls like `https://site/en/api/methods` to work?

### Describe the solution you'd like

The main solution to this problem is the ability to override API root path.

### Describe alternatives you've considered

I try rename theme to _api and rewrite url from `/en/api/article` to `/en/_api/article`, but rewrite rules in next.config don't work with `/api` (or `/en/api`) source, only destination.

**Comment by icyJoseph** [comment]
Hi,

Have you tried with something like:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  async rewrites() {
    return {
      beforeFiles: [
        {
          source: "/:locale/api",
          destination: "/_api/:locale",
        },
      ],
    };
  },
};

module.exports = nextConfig;
```

Then I can go to `http://localhost:3000/en/api` and show the contents of `pages/_api/[locale]`
